### PR TITLE
feat(openchallenges): add caption showing undatedChallengeCounts in timeline plot

### LINKED ISSUE #2339: [Task] Inconsistency between the total number of challenges in bar plot and the number listed in the challenge search
### What product(s) is this feature for?

OpenChallenges

### Description

There are some challenges lacking a defined start date, which are excluded from the bar plot visualization. This leads the inconsistency between the total number of challenges represented in the 2023 bar plot and the number listed in the challenge search feature (which reflects the database total).  

To avoid confusing users, it is proposed that we incorporate a small, grey caption directly into the bar plot area, like `*An additional 50 challenges without start dates are not displayed in the plot`, or other approaches if it will fit better.

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by rrchai**
- closes #2339 

## Depends on
- #2354 
- ~~Grand challenges data is added in `main`~~

## Changelog

- add caption about the number of missing challenge due to lack of start date in the plot

## Preview
Screen Shot 2023-11-13 at 6 31 21 PM

> [!NOTE]  
> The text in the preview has been updated to "*An additional 50 challenges without known start dates are not displayed."
> The static value for the challenge without start date will be update with DB update as well.

### CODE REVIEW

File: apps/openchallenges/challenge-service/src/main/resources/openapi.yaml [hunk 1]
```diff
@@ -891,8 +892,13 @@ components:
           items:
             type: integer
           type: array
+        undatedChallengeCount:
```
Comment (tschaffter) [inline apps/openchallenges/challenge-service/src/main/resources/openapi.yaml:895]: Specify a min value (0)

**Review by tschaffter — COMMENTED** [review]
See above comment

**Comment by tschaffter** [comment]
@rrchai The challenge service container works for me (see below).

If the container keeps restarting, you could try to remove the `build` folder of the service `./gradlew clean`, the rebuild the service BUT without relying on the Nx cache: `nx build openchallenges-challenge-service --skip-nx-cache`. I have feeling that it's this cache that may have been poisoned. Running this command should properly generate the `build` folder and update the cache. Only then you could then build the image again.

If the issue persist, you could rebuild the dev container to start fresh (remember to run the command `workspace-install` immediately after).

🤞

```console
HTTP/1.1 200 
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
X-Frame-Options: DENY
Content-Type: application/json
Transfer-Encoding: chunked
Date: Tue, 14 Nov 2023 02:58:44 GMT
Connection: close

{
  "years": [
    "2007",
    "2008",
    "2009",
    "2010",
    "2011",
    "2012",
    "2013",
    "2014",
    "2015",
    "2016",
    "2017",
    "2018",
    "2019",
    "2020",
    "2021",
    "2022",
    "2023"
  ],
  "challengeCounts": [
    5,
    8,
    12,
    16,
    21,
    27,
    31,
    38,
    45,
    54,
    80,
    91,
    110,
    129,
    177,
    203,
    226
  ],
  "undatedChallengeCount": 50
}
```

File: libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts [hunk 2]
```diff
@@ -75,10 +75,28 @@ export class StatisticsViewerComponent implements OnInit, OnDestroy {
...
             ],
+            graphic: res.undatedChallengeCount
+              ? {
+                  elements: [
+                    {
+                      type: 'text',
+                      style: {
+                        text:
+                          `*An additional ${res.undatedChallengeCount} challenges ` +
+                          `without start dates are not displayed in the plot`,
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts:91]: Font size too small? Compare with other font sizes we current use

[the remark this answers is not in the document]
Comment (rrchai) [reply libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts]: > Could we move this HTML code to the HTML file?

I was unable to locate any official docs about this transfer process. It must not be a straightforward way :).

**Comment by rrchai** [comment]
as @tschaffter suggests, I will separate caption part from plot itself and move caption code to html - converting to draft

File: libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts [hunk 3]
```diff
@@ -75,10 +75,29 @@ export class StatisticsViewerComponent implements OnInit, OnDestroy {
...
               },
             ],
+            graphic: res.undatedChallengeCount
+              ? {
+                  elements: [
+                    {
+                      type: 'text',
+                      style: {
+                        text:
+                          `*An additional ${res.undatedChallengeCount} challenges ` +
```
Comment (vpchung) [inline libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts:90]: It might just be me, but this reads awkward - I think because of the double negative?  Maybe reword it to:

"Due to missing start dates, an additional xx challenges are not displayed."

**Review by vpchung — APPROVED** [review]
LGTM. I do have one small suggestion to the language used.

File: libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts (hunk 3 above)
Comment (tschaffter) [reply libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts:90]: I had the same thought on Slack. Here are suggestions:

- *The OC database includes 50 more challenges without known start dates.
- *The OC database includes an extra 50 challenges without known start dates.

**Review by tschaffter — APPROVED** [review]
The PR looks good but I suggest rephrasing the caption to sound more positive (avoid double negative terms).

File: libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts (hunk 3 above)
Comment (rrchai) [reply libs/openchallenges/home/src/lib/statistics-viewer/statistics-viewer.component.ts:90]: Thanks both! I will replace text with`*The OC database includes an extra 50 challenges without known start dates.` for now~
